### ISSUE TITLE
support for element.isContentEditable

### ISSUE THREAD

**Issue by ianstormtaylor** [tmpvar/jsdom#1670]
It looks like the [`isContentEditable`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/isContentEditable) property is supported by all major browsers, but seems to be `undefined` when I try to access it in `jsdom`.

**Comment by domenic** [comment]
The question is, would it really be useful to have this feature given that we don't implement any of the rest of the content editable stuff?

**Comment by ianstormtaylor** [comment]
Fair question. 

I'd argue yes, because something like `node.isContentEditable` extends into realms where the content editing functionality does not. You'd expect having to fake the actual "editing" of the DOM when programmatically working with it, but even trying to do that becomes impossible is your pre-/post-editing logic depends on this property.

Not to even mention valid use cases that are looking for `isContentEditable == false` to avoid things.

**Comment by domenic** [comment]
I'm not really sure how to implement it though, if we don't implement content editable in general.

**Comment by ianstormtaylor** [comment]
I might be wrong since I don't know how JSDOM is implemented internally, but I'd assume it would be checking for attributes/elements that satisfy the ["editing host" or "editable"](https://w3c.github.io/editing/execCommand.html#editing-host) requirements, which mainly would involved looking up the DOM tree to see if its inside a `contenteditable` parent (or a few other cases).

**Comment by domenic** [comment]
Yeah, but what's weird then is that contenteditable="true" or contenteditable="false" have an impact, but that impact is only to change the value of isContentEditable, and not to actually let any of the content-editable behaviors (like execCommand and friends) work.

**Comment by snuggs** [comment]
@domenic just when I thought I knew the dom. You never cease to amaze me. `execCommand` = #mindBlown. How did I miss this?

**Comment by adamgruber** [comment]
Any plans for JSDOM to support `isContentEditable`? or is there a workaround?

**Comment by domenic** [comment]
No plans to implement contenteditability, but you can as usual monkeypatch the DOM as a workaround.

**Comment by dgoldstein0** [comment]
+1 to this feature request.  I am porting a bunch of tests from running in a browser to running in jest, and one of them needs contentEditable.  Basically we do

```
elem.contentEditable = 'true'
```

and then expect `elem.isContentEditable` to be true, as well as `elem.hasAttribute("contenteditable").toLowerCase() != "false"`.  Even if jsdom doesn't have full contentEditable support it would be helpful if it supported this much.

**Comment by domenic** [comment]
Sorry, we don't implement features that way. You're welcome to write a helper that makes the necessary monkey patching easy and use that in your tests though.

**Comment by dgoldstein0** [comment]
when you say you "don't implement features that way", you mean you don't like having half-implemented dom features?  or just that I shouldn't guess what dom behavior is supposed to be?

**Comment by domenic** [comment]
The former.

**Comment by dgoldstein0** [comment]
k, I'll keep that in mind.

**Comment by delijah** [comment]
+1

**Comment by masad-frost** [comment]
This issue is going start showing up more as more people pick up codemirror

**Comment by levensta** [comment]
This is also true for other modern editors such as Lexical and Tiptap, as they cannot be tested using jsdom
